### DISCUSSION TITLE
query by etag in 4&5?

### DISCUSSION THREAD

**Discussion by mirzahassan** [ravendb/ravendb#15350] [Q&A] [answered]
In 3.5, we use `StreamDocs(etag)` to get all docs that have changed since the last saved etag. 
Is there no equivalent in 4 or 5 to do this? 

I suspect Data Subs might be the answer, but as we are moving over our code base to 5.0, we wanted to see if this was an option or we have to rewrite.

**Comment by ayende** [comment, marked as answer]
No, there is no way to do that.
Basically, there is no way to say what is the order between documents written to on different nodes, so we didn't enable that.

Subscriptions is the way to go here.

Note that we still support the etag model, but that is mostly for the studio.

https://github.com/ravendb/ravendb/blob/v5.4/src/Raven.Server/Documents/Handlers/DocumentHandler.cs#L169

This is because from the API perspective, the server answering two requests may be different, and have different etags for the docs.
